### PR TITLE
doc: Dynamic Cross-Contract Calling

### PR DESCRIPTION
**Pull request by 0xf333**
## Description
- Updated ink! code snippets to conform to v4.
- Introduced working solidity equivalent to make the comparison easier to grasp.
- Removed comment regarding v3 limitations

### CODE REVIEW

File: docs/intro/ink-vs-solidity.md [hunk 1]
```diff
@@ -719,78 +719,102 @@ Now, to perform the cross-contract call:
...
+        }
+    }
+}
 ```
 
 Note: the `function_selector` bytes can be found in the generated `target/ink/<contract-name>.json`.
 
-## Limitations of ink! v3
-
-- Multi-file projects are not supported with pure ink!
```
Comment (SkymanOne) [inline docs/intro/ink-vs-solidity.md:787]: This is still the case

Comment (0xf333) [reply docs/intro/ink-vs-solidity.md:787]: Thanks for the feedback, requested limitation note added back.

**Comment by 0xf333** [comment]
@SkymanOne 

This should probably be fixed in a separated pull request since it's not part of this PR scope.
```
Check Markdown links / markdown-link-check (pull_request) 
```
Let me know what you think.

**Comment by 0xf333** [comment]
> @SkymanOne
> 
> This should probably be fixed in a separated pull request since it's not part of this PR scope.
> 
> ```
> Check Markdown links / markdown-link-check (pull_request) 
> ```
> 
> Let me know what you think.

@SkymanOne 
`markdown-link-check` failure has been fixed in #273 #272 #271

**Comment by 0xf333** [comment]
> @0xf333 Sorry for the very late response here. Are you up for merging `master` into your PR and resolving the conflicts?

@cmichi it's done.

**Note on CI Errors:**
While CI is showing search plugin configuration errors, this PR's scope is intentionally limited to resolving the merge conflict between `master branch` and PR #270.  This way we maintain clear separation of concerns to keep changes focused and traceable.

**Review by cmichi — APPROVED** [review]
Thank you, @0xf333 !

**Pull request merged**